workspaces/ui-v2/src/entry-points/cloud/cloud-viewer.tsx

```diff
@@ -49,10 +49,12 @@ const appConfig: OpticAppConfig = {
       sentryUrl: process.env.REACT_APP_SENTRY_URL,
     },
     backendApi: {
+      // Because we use the same docker image for both staging, and prod
+      // The UI needs to be context aware and know about _both_ staging and prod
       domain:
         window.location.hostname.indexOf('useoptic.com') >= 0
-          ? process.env.REACT_APP_PROD_API_BASE
-          : process.env.REACT_APP_STAGING_API_BASE,
+          ? 'https://spec.useoptic.com' // production
+          : 'https://api.o3c.info', // staging
```

Comment:
why hardcode these?

Reply:
ah, i read the full pr description. i see it called out

Reply:
This is ideally only going to be temporary. We have to determine these at run time (until we generate two artifacts).

The better solution here is to inject a single environment variable at build time and generate two artifacts (one for staging, one for production). I think it's also confusing to have `REACT_APP_API_BASE`, `REACT_APP_STAGING_API_BASE` and `REACT_APP_PROD_API_BASE` given that we share env variables across different builds (there's multiple entry points to inject environment variables, such as cloud viewer, building npm artifacts).

Reply:
If we want to be able to inject / control these, I can readd `REACT_APP_STAGING_API_BASE` and `REACT_APP_PROD_API_BASE` - I just wanted to avoid more confusion with our env variables, with the hope this is temporary.

Reply:
got it!

Reply:
@notnmeyer thoughts on staging+prod artifacts vs 1 artifact with logic to choose env? Either way both need to be tested, 2 artifacts means longer time and more room for error

Reply:
i thought the env was baked into application at build time isnt that the cruxt of the problem? if we can set the env on the container instead then i would absolutely prefer to do that.

Reply:
It's not possible to inject env variables in FE applications without some work (we could do something where we inject variables where we serve cloudviewer - but making it so we can use the same interface as `process.env` might be trickier). 

I strongly dislike having the UI having access to both environments, and rather these be injectable - if we let the UI have logic to handle both + select the env - there's a lot higher risk of cross contaimination between envs (debugging env errors would require looking in multiple places, wherever the UI implements this selection logic, and the env variable files (which now contain env variables for both).

Either way - this is a larger discussion. In respect to this PR - the hardcoded env variables here are intended to be temporary (there's also a bunch of other instances of hard coded env fallbacks in the UI - not that it justifies this) - any concerns with moving forward with the current implementation - we'll follow up with the staging / prod artifacts discussion separately @notnmeyer @devdoshi